File: leetcode/problems/py/find-the-highest-altitude.py
# by @codeAbinash
# Time : O(n)
# Space : O(1)

class Solution:
    def largestAltitude(self, gain: List[int]) -> int:
        altitude, max_altitude = 0, 0
        for g in gain:
            altitude += g
            max_altitude = max(max_altitude, altitude)
        return max_altitude

File: leetcode/problems/java/find-the-highest-altitude.java
// by @codeAbinash
// Time : O(n)
// Space : O(1)

class Solution {
  public int largestAltitude(int[] gain) {
    int altitude = 0, maxAltitude = 0;
    for (int i = 0; i < gain.length; i++) {
      altitude += gain[i];
      maxAltitude = Math.max(maxAltitude, altitude);
    }
    return maxAltitude;
  }
}